apps/indexer/src/api/services/treasury/providers/defillama.provider.ts

```diff
@@ -0,0 +1,166 @@
+import axios, { AxiosInstance } from "axios";
+import { TreasuryProvider } from "./treasury-provider.interface";
+import { TreasuryDataPoint, RawDefiLlamaResponse } from "../types";
+
+export class DefiLlamaProvider implements TreasuryProvider {
+  private readonly client: AxiosInstance;
+  private readonly providerDaoId: string;
+  private isValid: boolean = false;
+
+  private constructor(baseUrl: string, providerDaoId: string) {
+    this.client = axios.create({
+      baseURL: baseUrl,
+    });
+    this.providerDaoId = providerDaoId;
+  }
+
+  /**
+   * Factory method to create a validated provider instance
+   */
+  static async create(
+    baseUrl: string,
+    providerDaoId: string,
+  ): Promise<DefiLlamaProvider> {
+    const instance = new DefiLlamaProvider(baseUrl, providerDaoId);
+    await instance.validateConnection();
+    return instance;
+  }
+
+  /**
+   * Validates that the provider DAO ID exists in DeFi Llama.
+   * Logs warning if validation fails but doesn't throw.
+   */
+  private async validateConnection(): Promise<void> {
+    try {
+      console.log(
+        `[DefiLlamaProvider] Validating connection for DAO ID: ${this.providerDaoId}`,
+      );
+
+      const response = await this.client.get<RawDefiLlamaResponse>(
+        `/${this.providerDaoId}`,
+      );
+
+      // Check if response has valid data structure
+      if (
+        !response.data?.chainTvls ||
+        Object.keys(response.data.chainTvls).length === 0
+      ) {
+        throw new Error("Invalid response structure or empty data");
```

Comment:
should this be a throw or a log? it depends on how the system should behave when invalid envs are given